docker/Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM debian:bullseye-slim
+
+WORKDIR /web
+
+COPY . .
+
+RUN mv .env.example .env \
+    && mkdir -p media logs static/COMPILED .well-known \
+    && useradd -M -d /web -s /sbin/nologin uwsgi \
+    && printf 'APT::Install-%s false;\n' \
+        Recommends Suggests >/etc/apt/apt.conf.d/00minimal \
+    && printf 'APT::Keep-Downloaded-Packages false;\n' \
+        >>/etc/apt/apt.conf.d/00minimal \
+    && apt-get update \
+    && apt-get install -y \
+        libjemalloc2 \
+        python3-pip \
+        python3-psycopg2 \
+        uwsgi-plugin-python3 \
+    && pip install --no-cache-dir -e .[csp,sentry] \
+    && mv docker/uwsgi.ini /etc/uwsgi.ini \
+    && rm -rf docker /var/lib/apt/lists/* \
+    && ./manage.py collectstatic --no-input \
+    && chown -R uwsgi:uwsgi /web
+
+VOLUME /web/.well-known /web/media /web/logs
```

Comment:
You chown /web and create these volumes after. I think it might not be guaranteed that the uwsgi user will have access to them? I've run into a lot of stuff like this in the past. Speaking of, are you going to use the uwsgi user in this container? Might need a [USER](https://docs.docker.com/engine/reference/builder/#user) command to do this.

Usually VOLUME definition is done around the top of the file too for stylistic reasons but that's not really that big of a deal.

Reply:
I had to `chown` the directory for the `uwsgi` process to have proper access.

I don't think the `USER` command would achieve the same thing.
The docs don't mention if it applies to file ownership.

Reply:
My point was that the CMD there as it is rn will run as root so if you intend to use the uwsgi user to run it you should define it with USER. But apparently you define this in uwsgi.ini so that's fine.

Then the other point was that when you declare the VOLUMEs, this will create the mount point but just because you put them under a directory that uwsgi owns (/web) doesn't mean uwsgi will own the .well-known / media / logs folders at run time. Their permissions will depend 100% on what is set on the Docker host. 

It's up to you whether or not this is actually an issue but in another django/docker deployment I've used I've run into that exact problem with the media folder specifically. The way around this is to create the folders explicitly within the container and chown them. You will still be able to mount onto them with volume-related arguments in docker-compose or the docker CLI, but you guarantee that the container user will be able to use them properly.

Reply:
It does have access because they were created earlier.